backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/impl/BPartnerBL.java

```diff
@@ -531,6 +531,26 @@ public int getDiscountSchemaId(@NonNull final I_C_BPartner bpartner, final SOTrx
 			{
 				return groupDiscountSchemaId; // we are done
 			}
+
+			// didn't get the schema yet; now we try to get the discount schema from the parent C_BP_Group
+			final BPGroupId parentBpGroupId = BPGroupId.ofRepoIdOrNull(bpGroup.getParent_BP_Group_ID());
```

Comment:
why not doing it in a loop navigating the groups up to the root?

Reply:
it's basically 2 levels of G_BP_Group hierarchy that we're allowing, the first is references by the BP directly. So the loop would have at most 1 passes, for the first parent.